src/OrleansRuntime/Catalog/GrainCreator.cs

```diff
@@ -109,5 +96,10 @@ public Grain CreateGrainInstance(Type grainType, IGrainIdentity identity, Type s
 
             ((ILogConsistentGrain)grain).InstallAdaptor(factory, state, grainType.FullName, storageProvider, svc);
         }
+
+        public virtual void Release(IGrainActivationContext context, object grain)
```

Comment:
This was by design, since I expect most custom implementations will inherit from the default one and call the base in several occasions, similar to the one in our tests. This copies the DefaultControllerActivator from MVC